debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+        protected Schema operatingSchema(R record) {
+            return record.valueSchema();
+        }
+
+        @Override
+        protected Object operatingValue(R record) {
+            return record.value();
+        }
+
+        @Override
+        protected R newRecord(R record, Schema updatedSchema, Object updatedValue) {
+            return record.newRecord(record.topic(), record.kafkaPartition(), record.keySchema(), record.key(), updatedSchema, updatedValue, record.timestamp());
+        }
+    }
+
+    protected abstract Schema operatingSchema(R record);
+
+    protected abstract Object operatingValue(R record);
+
+    protected abstract R newRecord(R record, Schema updatedSchema, Object updatedValue);
+
+
+    /**
+     * Creates the new schema of the nested struct fields that matches with config.field_type.
+     * Assign optional fields to the fields that does not contain value or null value.
+     *
+     * @param field the input nested Struct field.
+     * @param value the input nested Struct field value.
+     **/
+    private SchemaBuilder NestedStructSchemaUpdate(Field field, Struct value) {
+        SchemaBuilder builder = SchemaUtil.copySchemaBasics(field.schema(), SchemaBuilder.struct());
+        for (Field structField : field.schema().fields()) {
+            String structFieldSchemaName = structField.schema().name();
+            final Object nestedStructField = ((Struct) value.get(field)).get(structField);
+            if (structFieldSchemaName != null && !structFieldSchemaName.isEmpty() && TypeConverters.containsKey(structFieldSchemaName)) {
+                if (nestedStructField == null || nestedStructField.equals("")) {
+                    builder.field(structField.name(), TRANSLATORS.get(TypeConverters.get(structFieldSchemaName)).optionalSchema());
+                } else {
+                    builder.field(structField.name(), TRANSLATORS.get(TypeConverters.get(structFieldSchemaName)).typeSchema());
+                }
+            } else {
+                if (nestedStructField == null || nestedStructField.equals("")) {
+                    builder.field(structField.name(), OPTIONAL_TRANSLATORS.get(structField.schema().type().toString()).optionalSchemaNonTimestamp());
+                } else {
+                    builder.field(structField.name(), structField.schema());
+                }
+            }
+        }
+        return builder;
+    }
+
+    /**
+     * Creates the new field value of the nested struct that matches with config.field_type.
+     * Avoids all the null value fields and converts the desired field value that matches to the config parameter field_type.
+     *
+     * @param field               the input nested Struct field.
+     * @param value               the input nested Struct field value.
+     * @param nestedUpdatedSchema updated field schema with target formats.
+     */
+    private Struct NestedStructFieldValueUpdate(Field field, Struct value, Field nestedUpdatedSchema) {
```

Comment:
Same here with lowercase first character and a more meaningful name (e.g., `valueForNestedStruct(...)`).